pkg/cmd/cli/cmd/whoami.go

```diff
@@ -50,3 +54,39 @@ func (o WhoAmIOptions) WhoAmI() (*userapi.User, error) {
 
 	return me, err
 }
+
+func RunWhoAmI(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args []string, o *WhoAmIOptions) error {
+	if kcmdutil.GetFlagBool(cmd, "token") {
+		cfg, err := f.OpenShiftClientConfig.ClientConfig()
+		if err != nil {
+			return err
+		}
+		if len(cfg.BearerToken) == 0 {
+			return fmt.Errorf("no token is currently in use for this session")
+		}
+		fmt.Fprintf(out, "%s\n", cfg.BearerToken)
+		return nil
+	}
+	if kcmdutil.GetFlagBool(cmd, "context") {
```

Comment:
> There's no easy way to get it today - since user identity is somewhat tied to context i'd rather have the option for someone to check their context, since project is more about the project you're in.

Why do we think users care about contexts?  I'm hard pressed to think of a reason.  That was one of the reasons why ugly names seemed reasonable there.